### DISCUSSION TITLE
Change max tokens in python

### DISCUSSION THREAD

**Discussion by rosmur** [Blaizzy/mlx-vlm#158] [Q&A]
Any documentation available on how to increase the max tokens? Its defaulting to 100 seemingly.

**Comment by Blaizzy** [comment]
Just pass `--max-tokens`

**Reply by rosmur** [reply]
That's in the CLI. I'm asking how to do it in Python

**Reply by Blaizzy** [reply]
Pass it `max_tokens` via the `generate`.

https://github.com/Blaizzy/mlx-vlm/blob/main/mlx_vlm/utils.py#L1161-L1174

**Reply by rosmur** [reply]
Thanks. Is there a reason the default is such a small number?

**Reply by Blaizzy** [reply]
Not really. 

I think when I initially built this package I didn't want it to generate a large blob of text when I was testing.

I can increase it. How much would you like to be the default?

**Reply by rosmur** [reply]
thanks, I think 2048 or 4096 would be good. Those are typical settings I've encountered for `generate` functions in other packages

**Reply by Blaizzy** [reply]
that's too high 

Most models don't output that much text.

How about 1000?

**Reply by rosmur** [reply]
yeah, I think thats ok. I've seen minimum context length defaults of 2k but those are for chat completion clients where naturally the context length increases with chat turns, but since this is a generate function 1000 makes sense
